pkg/networking/vpc_info_provider.go

```diff
@@ -0,0 +1,86 @@
+package networking
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	awssdk "github.com/aws/aws-sdk-go/aws"
+	ec2sdk "github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/go-logr/logr"
+	"github.com/patrickmn/go-cache"
+	"sigs.k8s.io/aws-load-balancer-controller/pkg/aws/services"
+)
+
+// VPCInfoProvider is responsible for providing VPC info.
+type VPCInfoProvider interface {
+	FetchVPCInfo(ctx context.Context, vpcID string) (*ec2sdk.Vpc, error)
+}
+
+// NewDefaultVPCInfoProvider constructs new defaultVPCInfoProvider.
+func NewDefaultVPCInfoProvider(cacheDuration int, ec2Client services.EC2, logger logr.Logger) *defaultVPCInfoProvider {
+	return &defaultVPCInfoProvider{
+		ec2Client:         ec2Client,
+		vpcInfoCache:      cache.New(time.Duration(cacheDuration)*time.Minute, 10*time.Minute),
```

Comment:
let's use the cache library from apimachinary, which automatically GC expired entries.
this can help us minimize the dependencies we have.

sample:
new cache: https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/pkg/networking/node_eni_info_resolver.go#L31
read cache: https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/pkg/networking/node_eni_info_resolver.go#L91
set cache: https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/pkg/networking/node_eni_info_resolver.go#L108